lib/cocoapods-core/cdn_source.rb

```diff
@@ -219,14 +219,21 @@ def search(query)
     #         the search term. Can be a regular expression.
     #
     # @param  [Bool] full_text_search
-    #         not supported due to performance reasons
+    #         performed using Algolia
     #
     # @note   full text search requires to load the specification for each pod,
     #         and therefore not supported.
     #
     def search_by_name(query, full_text_search = false)
       if full_text_search
-        raise Informative, "Can't perform full text search, it will take forever"
+        require 'algoliasearch'
+        begin
+          algolia_result = algolia_search_index.search(query, :attributesToRetrieve => 'name')
+          names = algolia_result['hits'].map { |r| r['name'] }
+          names.map { |n| set(n) }.reject { |s| s.versions.compact.empty? }
+        rescue Algolia::AlgoliaError => e
+          raise Informative, "CDN: #{name} - Cannot perform full-text search because Algolia returned an error: #{e}"
```

Comment:
should we expose "Algolia returned an error"? Maybe "Server returned an error"?

Reply:
Let's say that once we have more than one implementation, this message will be easy to change 🙃